### PR TITLE
apply cosmos sdk v0.50.x

### PR DESCRIPTION
**Pull request by yoosah**
Closes: #XXX

## Description

______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: rpc/backend/call_tx_test.go [hunk 1]
```diff
@@ -297,15 +297,15 @@ func (suite *BackendTestSuite) TestSendRawTransaction() {
 			common.Hash{},
 			false,
 		},
-		{
+		/*{
```
Comment (nowooj) [inline rpc/backend/call_tx_test.go:300]: Is there any reason for pass?

File: x/feemarket/keeper/params.go [hunk 2]
```diff
@@ -18,15 +18,18 @@ package keeper
...
 	sdk "github.com/cosmos/cosmos-sdk/types"
 )
 
 // GetParams returns the total set of fee market parameters.
 func (k Keeper) GetParams(ctx sdk.Context) (params types.Params) {
-	store := ctx.KVStore(k.storeKey)
-	bz := store.Get(types.ParamsKey)
+	store := k.storeService.OpenKVStore(ctx)
+	// TODO error handling
+	bz, _ := store.Get(types.ParamsKey)
```
Comment (nowooj) [inline x/feemarket/keeper/params.go:32]: error handling

File: x/feemarket/keeper/keeper_test.go [hunk 3]
```diff
@@ -122,20 +127,20 @@ func (suite *KeeperTestSuite) SetupApp(checkTx bool) {
...
+	acc.AccountNumber = suite.app.AccountKeeper.NextAccountNumber(suite.ctx)
 	suite.app.AccountKeeper.SetAccount(suite.ctx, acc)
 
 	valAddr := sdk.ValAddress(suite.address.Bytes())
-	validator, err := stakingtypes.NewValidator(valAddr, priv.PubKey(), stakingtypes.Description{})
+	validator, err := stakingtypes.NewValidator(valAddr.String(), priv.PubKey(), stakingtypes.Description{})
 	require.NoError(t, err)
-	validator = stakingkeeper.TestingUpdateValidator(suite.app.StakingKeeper, suite.ctx, validator, true)
-	err = suite.app.StakingKeeper.Hooks().AfterValidatorCreated(suite.ctx, validator.GetOperator())
+	err = suite.app.StakingKeeper.SetValidatorByConsAddr(suite.ctx, validator)
```
Comment (nowooj) [inline x/feemarket/keeper/keeper_test.go:136]: It looks same to below
https://github.com/xpladev/ethermint/pull/14/files#diff-17c4e5810e38949d62886789d63a0de3c1ea03be8dafef51da8e9c6dc95a9302R139

File: x/feemarket/keeper/keeper.go [hunk 4]
```diff
@@ -74,15 +80,17 @@ func (k Keeper) Logger(ctx sdk.Context) log.Logger {
 // SetBlockGasWanted sets the block gas wanted to the store.
 // CONTRACT: this should be only called during EndBlock.
 func (k Keeper) SetBlockGasWanted(ctx sdk.Context, gas uint64) {
-	store := ctx.KVStore(k.storeKey)
+	store := k.storeService.OpenKVStore(ctx)
 	gasBz := sdk.Uint64ToBigEndian(gas)
-	store.Set(types.KeyPrefixBlockGasWanted, gasBz)
+	// TODO error handling
+	_ = store.Set(types.KeyPrefixBlockGasWanted, gasBz)
```
Comment (nowooj) [inline x/feemarket/keeper/keeper.go:86]: error handling 

- https://github.com/xpladev/ethermint/pull/14/files#diff-5884196984ba136a64f23e20947dc3eb4f20e6ebc569b70f469b9791dd105e48R85-R86
- https://github.com/xpladev/ethermint/pull/14/files#diff-5884196984ba136a64f23e20947dc3eb4f20e6ebc569b70f469b9791dd105e48R92-R93
- https://github.com/xpladev/ethermint/pull/14/files#diff-5884196984ba136a64f23e20947dc3eb4f20e6ebc569b70f469b9791dd105e48R130-R131

File: x/feemarket/keeper/integration_test.go [hunk 5]
```diff
@@ -234,9 +234,6 @@ var _ = Describe("Feemarket", func() {
 						// Note that max priority fee per gas can't be higher than the max fee per gas (gasFeeCap), i.e. 30_000_000_000)
 						return txParams{nil, big.NewInt(minGasPrices - 10_000_000_000), big.NewInt(30_000_000_000), &ethtypes.AccessList{}}
 					}),
-					Entry("dynamic tx with GasFeeCap > MinGasPrices, EffectivePrice < MinGasPrices", func() txParams {
```
Comment (nowooj) [inline x/feemarket/keeper/integration_test.go:237]: Is there any reason to delete?

File: x/evm/statedb/statedb.go [hunk 6]
```diff
@@ -85,6 +85,11 @@ func (s *StateDB) Keeper() Keeper {
 	return s.keeper
 }
 
+// GetContext returns the transaction Context.
+func (s *StateDB) GetContext() sdk.Context {
```
Comment (nowooj) [inline x/evm/statedb/statedb.go:89]: I'm curious as to why it was added.

File: x/evm/keeper/utils_test.go [hunk 7]
```diff
@@ -439,7 +440,7 @@ func (suite *KeeperTestSuite) TestVerifyFeeAndDeductTxCostsFromUserBalance() {
 			expectPassDeduct: true,
 			from:             suite.address.String(),
 			malleate: func() {
-				suite.ctx = suite.ctx.WithIsCheckTx(true)
+				suite.ctx = suite.ctx.WithIsCheckTx(true).WithConsensusParams(*app.DefaultConsensusParams)
```
Comment (nowooj) [inline x/evm/keeper/utils_test.go:443]: It seems unnecessary.

Comment (nowooj) [reply x/evm/keeper/utils_test.go:443]: Same 
- https://github.com/xpladev/ethermint/pull/14/files#diff-86a2788c6f900e3e8d070486fa1b357f309c85c581bb1b7b64597b7d7445cc8dR126
- https://github.com/xpladev/ethermint/pull/14/files#diff-86a2788c6f900e3e8d070486fa1b357f309c85c581bb1b7b64597b7d7445cc8dR149

Comment (nowooj) [reply x/evm/keeper/utils_test.go:443]: Same
- https://github.com/xpladev/ethermint/pull/14/files#diff-63855f036ffb07403867a5ce483596b316680c10e6034fe60304cc9a787be380R912
- https://github.com/xpladev/ethermint/pull/14/files#diff-63855f036ffb07403867a5ce483596b316680c10e6034fe60304cc9a787be380R1097
- https://github.com/xpladev/ethermint/pull/14/files#diff-63855f036ffb07403867a5ce483596b316680c10e6034fe60304cc9a787be380R1205

File: x/feemarket/keeper/keeper_test.go (hunk 3 above)
Comment (nowooj) [reply x/feemarket/keeper/keeper_test.go:136]: Is there a reason why you removed `TestingUpdateValidator` and the `Hooks`?

File: app/upgrades.go [hunk 8]
```diff
@@ -78,16 +79,18 @@ func (app *EthermintApp) RegisterUpgradeHandlers(cdc codec.BinaryCodec, clientKe
 	}
 
 	baseAppLegacySS := app.ParamsKeeper.Subspace(baseapp.Paramspace).WithKeyTable(paramstypes.ConsensusParamsKeyTable())
-	app.UpgradeKeeper.SetUpgradeHandler(planName, func(ctx sdk.Context, plan upgradetypes.Plan, fromVM module.VersionMap) (module.VersionMap, error) {
+	app.UpgradeKeeper.SetUpgradeHandler(planName, func(ctx context.Context, plan upgradetypes.Plan, fromVM module.VersionMap) (module.VersionMap, error) {
+		// XXX WA for build
```
Comment (nowooj) [inline app/upgrades.go:83]: Remove XXX

Nit. According to the convention of other codes, `goCtx context.Context` and `ctx sdk.Context` are often used.

File: x/feemarket/module.go [hunk 9]
```diff
@@ -144,15 +144,14 @@ func (am AppModule) RegisterServices(cfg module.Configurator) {
 }
```
Comment (nowooj) [inline x/feemarket/module.go:145]: Add to 5 interface 
```
_ module.HasGenesis         = AppModule{}
_ module.HasServices        = AppModule{}
_ appmodule.HasBeginBlocker = AppModule{}
_ appmodule.HasEndBlocker   = AppModule{}
_ module.HasConsensusVersion = AppModuleBasic{}
```

And we can remove unused fucntion. 
RegisterRESTRoutes, RegisterInvariants, RegisterStoreDecoder, GenerateGenesisState, WeightedOperations

File: x/evm/module.go [hunk 10]
```diff
@@ -39,8 +41,10 @@ import (
 )
 
 var (
-	_ module.AppModule      = AppModule{}
-	_ module.AppModuleBasic = AppModuleBasic{}
+	_ module.AppModule          = AppModule{}
+	_ module.AppModuleBasic     = AppModuleBasic{}
+	_ appmodule.HasEndBlocker   = AppModule{}
+	_ appmodule.HasBeginBlocker = AppModule{}
 )
```
Comment (nowooj) [inline x/evm/module.go:48]: Add 3 interface
```
_ module.HasGenesis          = AppModule{}
_ module.HasConsensusVersion = AppModuleBasic{}
_ module.HasServices         = AppModule{}
```

And we can remove unused func.
RegisterInvariants, RegisterStoreDecoder, GenerateGenesisState, WeightedOperations

File: x/evm/module.go [hunk 11]
```diff
@@ -58,7 +62,7 @@ func (AppModuleBasic) RegisterLegacyAminoCodec(cdc *codec.LegacyAmino) {
 
 // ConsensusVersion returns the consensus state-breaking version for the module.
 func (AppModuleBasic) ConsensusVersion() uint64 {
-	return 5
```
Comment (nowooj) [inline x/evm/module.go]: Why did the version go up?

File: x/evm/client/cli/query.go [hunk 12]
```diff
@@ -17,11 +17,11 @@ package cli
```
Comment (nowooj) [inline x/evm/client/cli/query.go:17]: Use autocli, instead of cli/query define

File: rpc/backend/call_tx_test.go (hunk 1 above)
Comment (nowooj) [reply rpc/backend/call_tx_test.go:300]: https://github.com/xpladev/ethermint/pull/14/commits/e2cc87462bed5906b75418652bd0d54437947eaa

File: x/evm/keeper/keeper.go [hunk 13]
```diff
@@ -362,7 +362,10 @@ func (k Keeper) getBaseFee(ctx sdk.Context, london bool) *big.Int {
 
 // GetMinGasMultiplier returns the MinGasMultiplier param from the fee market module
 func (k Keeper) GetMinGasMultiplier(ctx sdk.Context) sdkmath.LegacyDec {
-	fmkParmas := k.feeMarketKeeper.GetParams(ctx)
+	fmkParmas, err := k.feeMarketKeeper.GetParams(ctx)
+	if err != nil {
+		panic(err)
```
Comment (nowooj) [inline x/evm/keeper/keeper.go]: This should be error handling, not panic.

File: x/feemarket/keeper/params.go [hunk 14]
```diff
@@ -63,13 +65,19 @@ func (k Keeper) SetParams(ctx sdk.Context, params types.Params) error {
 
 // GetBaseFeeEnabled returns true if base fee is enabled
 func (k Keeper) GetBaseFeeEnabled(ctx sdk.Context) bool {
-	params := k.GetParams(ctx)
+	params, err := k.GetParams(ctx)
+	if err != nil {
+		panic(err)
```
Comment (nowooj) [inline x/feemarket/keeper/params.go]: This should be error handling, not panic.

File: x/feemarket/keeper/params.go [hunk 15]
```diff
@@ -63,13 +65,19 @@ func (k Keeper) SetParams(ctx sdk.Context, params types.Params) error {
...
+	}
 	return !params.NoBaseFee && ctx.BlockHeight() >= params.EnableHeight
 }
 
 // GetBaseFee gets the base fee from the store
 func (k Keeper) GetBaseFee(ctx sdk.Context) *big.Int {
-	params := k.GetParams(ctx)
+	params, err := k.GetParams(ctx)
+	if err != nil {
+		panic(err)
```
Comment (nowooj) [inline x/feemarket/keeper/params.go]: This should be error handling, not panic.

File: x/feemarket/keeper/params.go (hunk 2 above)
Comment (nowooj) [reply x/feemarket/keeper/params.go:32]: https://github.com/xpladev/ethermint/pull/14/commits/c4b3e581fca67e61210ba8292adda0976a3e0c8f

File: x/evm/keeper/utils_test.go (hunk 7 above)
Comment (nowooj) [reply x/evm/keeper/utils_test.go:443]: https://github.com/xpladev/ethermint/pull/14/commits/0dd3e83b3d83abc20727e63bc0dcb5f5c3703dde

File: x/feemarket/module.go (hunk 9 above)
Comment (nowooj) [reply x/feemarket/module.go:145]: https://github.com/xpladev/ethermint/pull/14/commits/3f68bb88b9a78e7d3b3eca175997bb81f29fc03d
https://github.com/xpladev/ethermint/pull/14/commits/0dd3e83b3d83abc20727e63bc0dcb5f5c3703dde

File: x/evm/module.go (hunk 10 above)
Comment (nowooj) [reply x/evm/module.go:48]: https://github.com/xpladev/ethermint/commit/3f68bb88b9a78e7d3b3eca175997bb81f29fc03d
https://github.com/xpladev/ethermint/commit/0dd3e83b3d83abc20727e63bc0dcb5f5c3703dde

File: x/evm/module.go (hunk 11 above)
Comment (nowooj) [reply x/evm/module.go]: https://github.com/xpladev/ethermint/pull/14/commits/5829d3092507eb853b29c69802adc981a3b043fd

File: x/evm/statedb/statedb.go (hunk 6 above)
Comment (nowooj) [reply x/evm/statedb/statedb.go:89]: https://github.com/xpladev/ethermint/pull/14/commits/0dd3e83b3d83abc20727e63bc0dcb5f5c3703dde

File: app/upgrades.go (hunk 8 above)
Comment (nowooj) [reply app/upgrades.go:83]: https://github.com/xpladev/ethermint/pull/14/commits/0dd3e83b3d83abc20727e63bc0dcb5f5c3703dde

File: x/feemarket/keeper/keeper_test.go (hunk 3 above)
Comment (nowooj) [reply x/feemarket/keeper/keeper_test.go:136]: https://github.com/xpladev/ethermint/pull/14/commits/1329587c9fd2de40733756cb1f10324574e4d953

File: app/ante/eth.go [hunk 16]
```diff
@@ -309,7 +309,10 @@ func (ctd CanTransferDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simulate
 		}
 
 		stateDB := statedb.New(ctx, ctd.evmKeeper, statedb.NewEmptyTxConfig(common.BytesToHash(ctx.HeaderHash())))
-		evm := ctd.evmKeeper.NewEVM(ctx, coreMsg, cfg, evmtypes.NewNoOpTracer(), stateDB)
+		evm, err := ctd.evmKeeper.NewEVM(ctx, coreMsg, cfg, evmtypes.NewNoOpTracer(), stateDB)
+		if err != nil {
+			return ctx, err
```
Comment (nowooj) [inline app/ante/eth.go]: Looks like it need a wrap with errorsmod.

File: x/evm/keeper/keeper.go (hunk 13 above)
Comment (nowooj) [reply x/evm/keeper/keeper.go]: https://github.com/xpladev/ethermint/pull/14/commits/7d4ec44e9fbe4243034d52548670ea13eb597878

File: x/feemarket/keeper/params.go (hunk 14 above)
Comment (nowooj) [reply x/feemarket/keeper/params.go]: https://github.com/xpladev/ethermint/pull/14/commits/3914a350d0e6bb6fca5a4faa27dd4efe823c7e2a

File: x/feemarket/keeper/params.go (hunk 15 above)
Comment (nowooj) [reply x/feemarket/keeper/params.go]: https://github.com/xpladev/ethermint/pull/14/commits/3914a350d0e6bb6fca5a4faa27dd4efe823c7e2a

File: x/feemarket/keeper/keeper.go (hunk 4 above)
Comment (nowooj) [reply x/feemarket/keeper/keeper.go:86]: https://github.com/xpladev/ethermint/pull/14/commits/cbeb6c95cabd13f28d8ad98328ddc210394c298f

File: x/evm/client/cli/query.go (hunk 12 above)
Comment (yoosah) [reply x/evm/client/cli/query.go:17]: I'll check this later

File: app/ante/eth.go (hunk 16 above)
Comment (nowooj) [reply app/ante/eth.go]: https://github.com/xpladev/ethermint/pull/14/commits/57497dde9ee668ed5e95bc3bec40fae639c1b1a4

File: x/feemarket/keeper/integration_test.go (hunk 5 above)
Comment (nowooj) [reply x/feemarket/keeper/integration_test.go:237]: https://github.com/xpladev/ethermint/pull/14/commits/9a354f1e613aec65403d70ff899b758080d9060d

File: x/evm/client/cli/query.go (hunk 12 above)
Comment (nowooj) [reply x/evm/client/cli/query.go:17]: https://github.com/xpladev/ethermint/issues/15

**Review by nowooj — APPROVED** [review]

**Pull request merged**